### DISCUSSION TITLE
Double press on tap

### DISCUSSION THREAD

**Discussion by Kaijonsson** [rnmapbox/maps#2764] [Ideas]
## New Feature
I would like to be able to disable the "double-press-zooming" feature.

## Why
 The delayed action it provides makes the UI i buildt to select shapesources is pretty anoying.

**Comment by pablogdcr** [comment]
Hi @Kaijonsson,

I'm facing the same issue. Even with the "double-press-zooming" feature, it should not take that long to trigger the press event.

I wasn't able to fix it yet, but as a workaround, you can take a look at the comment I wrote on this issue: https://github.com/rnmapbox/maps/issues/539#issuecomment-1443758905

**Comment by pablogdcr** [comment]
> Hi @Kaijonsson,
> 
> I'm facing the same issue. Even with the "double-press-zooming" feature, it should not take that long to trigger the press event.
> 
> I wasn't able to fix it yet, but as a workaround, you can take a look at the comment I wrote on this issue: [#539 (comment)](https://github.com/rnmapbox/maps/issues/539#issuecomment-1443758905)

This workaround isn't working anymore since `10.0.0-rc.2`.
It seems that `onTouchStart`, and `onTouchEnd` are not fired anymore.

**Comment by mfazekas** [comment]
On iOS it can be implemented with GestureOptions:  https://docs.mapbox.com/ios/maps/api/10.12.0/Structs/GestureOptions.html

Something like:

```swift
mapView.gestures.options.quickZoomEnabled = false
```

**Reply by Kaijonsson** [reply]
thing is, at least on my end. I wouldn't want to completely remove the double-tap. 

It's just when rendering a map that you can interact with, the delay for me when pressing a point and getting the answer is really bad for the UX. 

We are rendering a map where you can select/unselect fields that you own. And when pressing a field, marking it as yours, it's maybe 200 milliseconds that i have to wait before the action is recorded. 

But having the double tap on other maps in the app is really nice for the UX aswell! ;)
I've noticed that there is a prop in the web-client for mapbox to disable "double-click".

**Comment by ryanpwaldon** [comment]
Any workarounds that don't require editing native code? For my use case, I'd be happy removing double-tap zoom if it meant I could speed up onPress events.
